chia/rpc/full_node_rpc_api.py

```diff
@@ -399,6 +404,32 @@ async def get_block_records(self, request: Dict) -> EndpointResult:
             records.append(record)
         return {"block_records": records}
 
+    async def get_block_spends(self, request: Dict) -> EndpointResult:
+        if "header_hash" not in request:
+            raise ValueError("No header_hash in request")
+        header_hash = bytes32.from_hexstr(request["header_hash"])
+        full_block: Optional[FullBlock] = await self.service.block_store.get_full_block(header_hash)
+        if full_block is None or full_block.transactions_generator is None:
+            raise ValueError(f"Block {header_hash.hex()} not found or invalid block generator")
+
+        spends: List[CoinSpend] = []
+        block_generator = await self.service.blockchain.get_block_generator(full_block)
+        if block_generator is None:
+            return {"block_spends": spends}
+
+        args = create_generator_args(block_generator.generator_refs).first()
+        _, block_result = block_generator.program.run_with_cost(self.service.constants.MAX_BLOCK_COST_CLVM, 0, args)
```

Comment:
this has to be run through the generator "rom" program. see here: https://github.com/Chia-Network/chia-blockchain/blob/main/chia/full_node/mempool_check_conditions.py#L55

Reply:
Thanks

Reply:
Made a custom program as that program did not output what i needed, please take a look,